Task/hash password

adael2 opened the pull request:
#### What's this PR do?

Hash the password of the user

#### Where should the reviewer start?

express-app/controllers/user.js
express-app/models/user.js

#### How should this be manually tested?

After you login your password is saved with some caracters that hide the original caracters

#### Any background context you want to provide?



This template was adapted ~retrived~ from [Quickleft/Sprint.ly](https://quickleft.com/blog/pull-request-templates-make-code-review-easier/)

Commit:
Hash the password of user

This commit hash the password of user. For more sucurity.

Commit:
Merge branch 'develop' into task/encryption

Hashing password and password reset

Commit:
Delete TestDB

Ral15 reviewed (changes requested):
The order of the login should be:
1. Check if there is internet (online function)
2. IF there is internet check with the API.
3. IF username and password are OK, check in DB for the username and update the password (this in case the username changes the password in the online app). If there is NO username in the DB create the user and save it.
4. IF there is NO internet check directly in the DB.

erikiado commented:
Just like the raul said, step 3 is inside step 2 and step 4 is the else of step 2

erikiado commented on express-app/controllers/user.js:
```diff
@@ -24,16 +26,24 @@ module.exports = {
     // Call promise that validates internet connection
     isOnline().then((online) => {
       // look for the user in the local db first
```
This changed order, check the comment raul made

Commit:
Changes requested.

Check if there is internet (online function)
IF there is internet check with the API.
IF there is NO internet check directly in the DB.

Commit:
changes requested.

Reset the password from users.

erikiado commented on express-app/controllers/user.js:
```diff
@@ -109,4 +127,21 @@ module.exports = {
+  */
+  resetPassword: function (user, data) {
+    let hash = bcrypt.hashSync(data.password, bcrypt.genSaltSync(10), null);
+    User.findOneAndUpdate({ _id: user._id }, { password: hash }).then((editedUser) => {
```
Have you tested this at least manually, I'm not sure if when the password is reset, it would be hashed 2 times or not? This I think because you are hashing the password weach time the user is saved right?

adael2 replied:
yes, no it hash only one time  in preSave() is only when is save(). When is updated it doesn't enter in the function.

erikiado commented on express-app/controllers/user.js:
```diff
@@ -59,6 +66,7 @@ module.exports = {
    * @param {object} response - response object.
    */
   requestUser: function (data, request, response) {
+    var self = this;
```
Why not use just this?

adael2 replied:
Because the configuration change and if a used this the application send an error. Thats why use self insted this.

Commit:
Merge branch 'develop' into task/hash-password

Commit:
User password save only the first one and when it's reset.
